### PR TITLE
use json to structure scrolly steps

### LINKED ISSUE #10: generate scroll steps from input json
This has a few parts.
1. want a master json file that outlines the scroll sequence, including unique ids for each step, frames, and corresponding text. 
2. set vue template to `v-for` through the json structure and create scrolly divs and stacked images
3. apply gsap triggers to scrolly divs and images for transitions

### PR DESCRIPTION
**Pull request by cnell-usgs**
- Moves scrolling text for separate json `scrollyText.js` and uses `import` within `DroughtThreshold.vue` to read in. Addresses pt 1 https://github.com/USGS-VIZLAB/what-is-drought/issues/10
- Does minor clean up of unused components and `referencesText.js`
- Applies `v-for` directive to dynamically source images from `scrollyText.js`. Addresses pt 2 https://github.com/USGS-VIZLAB/what-is-drought/issues/10
- uses `forEach` to apply scroll trigger to to each step. this is driven by the `<div>` elements generated from the json text Addresses pt 3 https://github.com/USGS-VIZLAB/what-is-drought/issues/10

closes https://github.com/USGS-VIZLAB/what-is-drought/issues/10

### CODE REVIEW

File: src/components/DroughtThresholds.vue [hunk 1]
```diff
@@ -142,91 +66,26 @@ text: 'This variable threshold changes the timing and number of droughts over th
       // find all scrolly divs
       const containers = this.$gsap.utils.toArray(".scrolly");
 
-      // loop through and add scroll trigger to timeline for each step (not working)
+      //  add scroll trigger to timeline for each step
       containers.forEach((container) => {
-        /* tl.to(container, {
+
+        // get unique ID and class for frame
```
Comment (hcorson-dosch-usgs) [inline src/components/DroughtThresholds.vue]:
```suggestion
        // get unique ID and class for frame. Scroll frame classes follow the pattern `scrolly scroll-step-${frame.id}`
```

File: src/components/DroughtThresholds.vue [hunk 2]
```diff
@@ -142,91 +66,26 @@ text: 'This variable threshold changes the timing and number of droughts over th
...
+
+      // use class to set trigger
+        tl.to(`.${scrollClass}`, {
           scrollTrigger: {
             markers: this.marker_on,
-            trigger: container,
+            trigger: `.${scrollClass}`,
             start: "top 50%",
-            toggleClass: {targets: ""}
+            toggleClass: {targets: `#step-${scrollID}`, className:"onTop"}, // adds class to target when triggered
```
Comment (hcorson-dosch-usgs) [inline src/components/DroughtThresholds.vue:84]: This is so clean and compact now!

Comment (hcorson-dosch-usgs) [reply src/components/DroughtThresholds.vue:84]: I think the only thing left to sort out is how best to trigger the `unstuck` class for the `#spacer` div. Looks like this was [done manually before](https://github.com/USGS-VIZLAB/what-is-drought/blob/main/src/components/DroughtThresholds.vue#L222-L229). 

As of now the 'references' section scrolls under the final chart frame

**Review by hcorson-dosch-usgs — APPROVED** [review]
Really nice edits, Cee. It makes this code so much more compact and efficient. And it makes it that much easier to slot in any given set of images and text!

I think the only thing that needs to be fixed at some point is that there is no longer a step to add the `unstuck` class to the `spacer` div, which means that the 'references' section scrolls underneath the final chart. I think that could be added as an issue, for now, if you want to go ahead and merge these code changes.

File: src/components/DroughtThresholds.vue (hunk 2 above)
Comment (cnell-usgs) [reply src/components/DroughtThresholds.vue:84]: ok will add this as an issue

**Pull request merged**